### PR TITLE
Add clearRetiredFederation test

### PR DESCRIPTION
**Pull request by julia-zack**
-Add clearRetiredFederation test

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2047,6 +2047,22 @@ void getLastRetiredFederationP2SHScript_afterRSKIP186_whenSavingScript_returnsSc
...
 
+    @Test
+    @Tag("clear retired federation")
+    void clearRetiredFederation_whenHavingOldFederation_removesOldFederation() {
+        ErpFederation federation = new P2shErpFederationBuilder().build();
+        storageProvider.setOldFederation(federation);
+
+        // check the old federation was correctly saved
+        Federation oldFederation = storageProvider.getOldFederation(federationMainnetConstants, mock(ActivationConfig.ForBlock.class));
+        assertThat(oldFederation, is(federation));
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2058]: not so sure of doing this... this has been already tested in `getOldFederation` tests

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -2047,6 +2047,22 @@ void getLastRetiredFederationP2SHScript_afterRSKIP186_whenSavingScript_returnsSc
...
     }
 
+    @Test
+    @Tag("clear retired federation")
+    void clearRetiredFederation_whenHavingOldFederation_removesOldFederation() {
+        ErpFederation federation = new P2shErpFederationBuilder().build();
+        storageProvider.setOldFederation(federation);
+
+        // check the old federation was correctly saved
+        Federation oldFederation = storageProvider.getOldFederation(federationMainnetConstants, mock(ActivationConfig.ForBlock.class));
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2057]: Move the mock activations into its own variables then use it in the two places where it's been needed

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 3]
```diff
@@ -2047,6 +2047,22 @@ void getLastRetiredFederationP2SHScript_afterRSKIP186_whenSavingScript_returnsSc
         assertThat(lastRetiredFederationP2SHScript.get(), is(p2shScript));
     }
 
+    @Test
+    @Tag("clear retired federation")
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2051]: What's the benefit of using tags in this way with specific values for each test?  According to the [documentation](https://junit.org/junit5/docs/5.0.2/api/org/junit/jupiter/api/Tag.html) the intended use is for group common types of test.

**Review by nathanieliov — APPROVED** [review]
Well done Juli! Just one suggestion and an open discussion regarding the use of the `Tag` annotation.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2057]: good one, thanks!

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2051]: tbh it was inertia jaja, i can remove it

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: done

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 3 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2051]: If it is in more places(prs already merged), let it be as it is for now, and let's discuss this with the rest of team later.

**Review by nathanieliov — APPROVED** [review]

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**